src/pages/ZNS/ZNS.tsx

```diff
@@ -0,0 +1,62 @@
+//- React Imports
+import { FC } from 'react';
+
+//- Hook Imports
+import { useSubdomainData } from '../../lib/hooks/useSubdomainData';
+import { useDomainData } from '../../lib/hooks/useDomainData';
+import { usePaymentTokenInfo } from '../../lib/hooks/usePaymentTokenInfo';
+import { usePaymentTokenForDomain } from '../../lib/hooks/usePaymentTokenForDomain';
+import { useDomainMetadata } from '../../lib/hooks/useDomainMetadata';
+import { useDomainMetrics } from '../../lib/hooks/useDomainMetrics';
+import { useViewNavigation } from '../../lib/hooks/useViewNavigation';
+
+//- Provider Imports
+import { ModalProvider } from '../../lib/providers/ModalProvider';
+
+//- Container Imports
+import SubdomainView from '../../containers/subdomain-view/SubdomainView';
+import NFTView from '../../containers/nft-view/NFTView';
+
+//- Utils Imports
+import { getDomainId } from '../../lib/util/domains/domains';
+
+type ZNSProps = {
+	route: string;
+};
+
+const ZNS: FC<ZNSProps> = ({ route }) => {
+	const domainId = getDomainId(route);
+	const { data: domain } = useDomainData(domainId);
+	const { data: subdomainData } = useSubdomainData(domainId);
+	const { data: paymentToken } = usePaymentTokenForDomain(domainId);
+	const { data: paymentTokenInfo } = usePaymentTokenInfo(paymentToken);
+	const { data: metrics } = useDomainMetrics(domainId);
+	const { data: domainMetadata } = useDomainMetadata(domain?.metadataUri);
+	const { isNFTView } = useViewNavigation(subdomainData);
+
+	return (
+		<ModalProvider>
```

Comment:
Added a ModalProvider that can contain each modal to be used anywhere in subdomain view or nft view.